src/main.cpp

```diff
@@ -194,7 +194,7 @@ TCPsocket* net_tcpclients = nullptr;
 SDLNet_SocketSet tcpset = nullptr;
 list_t safePacketsSent, safePacketsReceived[MAXPLAYERS];
 bool receivedclientnum = false;
-char* window_title = nullptr;
+char const * window_title = nullptr;
```

Comment:
Out of curiosity, any particular reason why this is being changed to `char const*`? I don't see what's wrong with letting the window_title string be modifiable (although I notice we don't ever directly modify it).

Reply:
The C standard (I can't speak to C++) specifies that a string initialized that way can be stored in the code segment. GCC (I don't know about other compilers, but they probably do similar) is smart enough to move the modifiable string out of the code page.

The reason having a modifiable string in the code segment is bad is, if the OS and hardware support the NX bit, modifying it will cause a segmentation violation (SIGSEGV) and crash the program.

Intending to modify it or not, making it a const is The Right Thing.

There may be other instances where this error occurs, but this one gave me a warning, so I fixed it.

Reply:
So just for my own learning, is the issue coming from this pointer, or more from that it could potentially point to something like a string literal (which I believe actually may be the case)?

Reply:
Yup. Points to a string literal, which may live in read-only memory, which when modified, would cause a segfault. Therefore we should tag it const.